tests/testthat.R

```diff
@@ -1,2 +1,2 @@
 library(testthat)
-test_check("rtables", reporter = JunitReporter$new(file = "unit_test_results.xml"))
+test_check("rtables", reporter = "check")
```

Comment:
You sure? Don't we have single-core runners in GHA? Then it could be even longer than the default one.

Reply:
I just realised that we are anyway replacing this in our check action ([click](https://github.com/insightsengineering/r.pkg.template/blob/main/.github/workflows/build-check-install.yaml#L298-L318)) so change here won't have impact in timings within that action (we would have to change it there - definitely worth checking this).
This highlighted line would have impact on testing locally ... and new integration tests that I am working on. Please give me a moment to confirm that it gives some informative error message so that the root issue would be addressed.

Reply:
I am sorry to say that the ParallelReporter don't give informative error message hence cannot be implemented

This is what I got using ParallelReporter:
```
(...)
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’
 ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
  ✔ |  36       | split_funs [1.5 s]                                              
  
  ✔ |   6       | structure-summary [0.5 s]                                       
  
  ✔ |  74       | subset-access [4.2 s]                                           
  
                                                                                  
  ══ Results ═════════════════════════════════════════════════════════════════════
  Duration: 83.6 s
  
  [ FAIL 1 | WARN 1 | SKIP 0 | PASS 722 ]
  Error: Test failures
  In addition: Warning message:
  replacing previous import 'vctrs::data_frame' by 'tibble::data_frame' when loading 'dplyr' 
  Execution halted
* checking for unstated dependencies in vignettes ... OK
(...)
```

Whereas the same state of code using "check" reporter:
```
* checking for unstated dependencies in ‘tests’ ... OK
* checking tests ...
  Running ‘testthat.R’
 ERROR
Running the tests in ‘tests/testthat.R’ failed.
Last 13 lines of output:
   1. └─formatters::export_as_rtf(tbl, file = tmpf) at test-exporters.R:251:4
   2.   └─base::lapply(...)
   3.     └─formatters (local) FUN(X[[i]], ...)
   4.       ├─r2rtf::rtf_encode(...)
   5.       │ └─r2rtf:::.rtf_encode_table(tbl)
   6.       │   ├─base::paste(...)
   7.       │   └─r2rtf:::.as_rtf_color(tbl)
   8.       │     └─r2rtf:::.color_used(tbl)
   9.       └─formatters::mpf_to_rtf(...)
  
  [ FAIL 1 | WARN 1 | SKIP 0 | PASS 722 ]
  Error: Test failures
  In addition: Warning message:
  replacing previous import 'vctrs::data_frame' by 'tibble::data_frame' when loading 'dplyr' 
  Execution halted
* checking for unstated dependencies in vignettes ... OK
```

Reply:
So it does not tell you where the error happens. Not the best for bug-fixing

Reply:
Checked this only locally but please see PR description. The whole discussion started from JunitReporter giving me no error message on runners. So if that happens for JunitReporter on runners then I would think it's the same for ParallelReporter